sysroot_creator/scripts/create_cc_sysroot.py

```diff
@@ -142,184 +147,160 @@ def __str__(self):
-                is not None)
-            sysroot_tar.extractall(str(target_sysroot_path),
-                                   members=relevant_members)
-    finally:
-        shutil.rmtree(tmp_sysroot_dir, ignore_errors=True)
-    logger.info('Success exporting sysroot to path [{}]'.format(target_sysroot_path))
-
-
-def write_cc_build_setup_file(cc_root_dir: Path, platform: Platform) -> Path:
-    """Create setup file for cross-compile build."""
-    cc_build_setup_file_path = cc_root_dir / 'cc_build_setup.bash'
-    cc_build_setup_file_contents = CC_BUILD_SETUP_FILE_TEMPLATE.render(
-        target_arch=platform.arch,
-        target_triple=platform.cc_toolchain,
-        cc_root=cc_root_dir,
-        ros_root='{cc_root_dir}/sysroot/opt/ros/{distro}'.format(
-            cc_root_dir=cc_root_dir, distro=platform.distro))
-    with open(str(cc_build_setup_file_path), 'w') as out_f:
-        out_f.write(cc_build_setup_file_contents)
-    return cc_build_setup_file_path
-
-
-def write_cc_system_setup_script(
-        cc_root_dir: Path, platform: Platform) -> Path:
-    """Create setup file for sysroot setup."""
-    cc_system_setup_script_path = cc_root_dir / 'cc_system_setup.bash'
-    cc_system_setup_script_contents = \
-        CC_BUILD_SYSTEM_SETUP_SCRIPT_TEMPLATE.render(
-            cc_root=cc_root_dir,
-            target_triple=platform.cc_toolchain
-        )
-    with open(str(cc_system_setup_script_path), 'w') as out_f:
-        out_f.write(cc_system_setup_script_contents)
-    return cc_system_setup_script_path
-
-
-def setup_sysroot_environment(
-        sys_setup_path: Path, build_setup_path: Path) -> None:
-    """Set up the environment with variables and symbolic links."""
-    logger.info('Sourcing sysroot environment...')
-    logger.info("Executing 'bash {}'".format(sys_setup_path))
-    subprocess.run(['bash', str(sys_setup_path)])
-    logger.info("Executing 'source {}'".format(build_setup_path))
-    subprocess.run(['source', str(build_setup_path)], shell=True)
+class SysrootCreator:
+    def __init__(
+      self,
+      cc_root_dir: str,
+      ros_workspace_dir: str,
+      platform: Platform,
+      docker_config: DockerConfig):
+        self._cc_root_dir = Path(cc_root_dir)
+        self._ros_workspace_dir = Path(ros_workspace_dir)
+        self._target_sysroot = self._cc_root_dir / SYSROOT_DIR_NAME
+        self._ros_ws_directory = self._target_sysroot / self._ros_workspace_dir
+        self._qemu_directory = self._target_sysroot / QEMU_DIR_NAME
+        self._dockerfile_directory = Path(__file__).parent / DOCKER_WS_NAME
+        self._expected_dockerfile_directory = self._target_sysroot / DOCKER_WS_NAME
+        self.system_setup_script_path = Path()
+        self.build_setup_script_path = Path()
+        self._platform = platform
+        self._docker_config = docker_config
+
+        try:
+            self._setup_sysroot_dir()
+        except FileNotFoundError as e:
+            print(e)
+            sys.exit(1)
```

Comment:
sounds like this should be done at the main level

Reply:
Whoops.... Missed that.